### ISSUE TITLE
api poster uploads [Request]

### ISSUE THREAD

**Issue by BitPyrate** [HDInnovations/UNIT3D-Community-Edition#3897]
We are starting have a thriving unit3d api scripting community that has found a home on our tracker. 

they are requesting api upload support for
Table:movies
column:poster
to link to a poster offsite or perhaps upload a poster 

suggestion:
perhaps a condition when tmdb and the other meta sources = 0

**Comment by z-ink** [comment]
Hi, ignorant to php code here: but as the [http controller](https://github.com/HDInnovations/UNIT3D-Community-Edition/blob/master/app/Http/Controllers/TorrentController.php) has provision in lines 227-247 and again in lines 463-483 

`     // Cover Image for No-Meta Torrents
        if ($request->hasFile('torrent-cover')) {
            $image_cover = $request->file('torrent-cover');

            abort_if(\is_array($image_cover), 400);

            $filename_cover = 'torrent-cover_'.$torrent->id.'.jpg';
            $path_cover = public_path('/files/img/'.$filename_cover);
            Image::make($image_cover->getRealPath())->fit(400, 600)->encode('jpg', 90)->save($path_cover);
        }

        // Banner Image for No-Meta Torrents
        if ($request->hasFile('torrent-banner')) {
            $image_cover = $request->file('torrent-banner');

            abort_if(\is_array($image_cover), 400);

            $filename_cover = 'torrent-banner_'.$torrent->id.'.jpg';
            $path_cover = public_path('/files/img/'.$filename_cover);
            Image::make($image_cover->getRealPath())->fit(960, 540)->encode('jpg', 90)->save($path_cover);
        }`
        
        

> The api however has no access to post a 'torrent-cover' or 'torrent-banner', categories that are unlinked to TMDb would greatly benefit from this. 

_Quoted above because it kept throwing it into the code block_

**Comment by Roardom** [comment]
The movie and tv tables are strictly linked to tmdb currently. You can change the meta on the category to "no meta", and then you can upload your own banner/poster and it will be saved on the individual torrent. If you find content missing on tmdb, I would recommend adding it so that others can benefit. Currently the unit3d api doesn't accept those parameters, but if it did, it would only be allowed for categories with "no meta" selected. That being said, a PR to add the missing functionality to the api of uploading poster/banner for torrents with the "no meta" category would be appreciated.

**Comment by BitPyrate** [comment]
The main point would be to upload to the "No meta" categories with a poster. right now if you mass upload to a no meta group you have to manually go back and add it. 
uploading a handful that's not a big problem however are faced with uploading thousands of no meta uploads. this is why im asking about the api provisions. If it was just myself one be one thing but we have several others facing the same dilemma.

We are aware about adding entries to tmdb however we are starting to have large backlogs for our upload scripts with non film stuff. book, audiobooks, we also have very obscure old films which for various reasons are not permitted to be added to tmdb. (i.e popeye 2013). TMDB also forbids many independent releases outside the studio systems or films that did not have a theatrical release.

Porn being a lesser concern as we prefer those to follow the rules closely and prefer porn to be uploaded manually if they are absent from tmdb.

I figured this would be eventually added so my personal backlog of thousands of titles i was holding off.  It was only after other users are starting to wish they could automate their "no meta" uploads better is what prompted this request.

while overriding existing tmdb poster would be cool for movie meta categories that is not as essential, but a cool stretch goal

I suspect all we need is the routes added, and the torrentcontroller adapted to handle it. I'm just not an expert in this area yet. Ill spend time on it if i can but i figured it might be a relatively easy add by people more familiar with this part of the code.

**Comment by HDVinnie** [comment]
I'll have this added for next release.

**Comment by z-ink** [comment]
> I'll have this added for next release.

Fantastic news! I do think it would be best in long run if image could be hosted elsewhere as in we send a url to direct image address.. but beggars cant be choosers so any kind of support right now is better than _none_

**Comment by mibj2** [comment]
Hello thank you for bringing me here
Please, I am trying to understand how to send the payload with a 'no meta' category. If I understand correctly, I need to use that category

Once a 'no meta' category is created, should I send the image URL through the 'tracker-cover' payload attribute or just the link?

My template for body data:
`
 self.headers = {
            "User-Agent": "Test/0.0 (Linux 5.10.0-23-amd64)",
            "Accept": "application/json",
        }
        self.params = {
            "api_token": self.api_token,
        }

        self.data = {
            "name": "TEST.torrent",
            "description": "",  # mandatory
            "mediainfo": "",
            "bdinfo": " ",
            "type_id": "1",
            "resolution_id": 10,  # mandatory
            "tmdb": "",  # mandatory
            "imdb": "0",  
            "tvdb": "0",  
            "mal": "0",  
            "igdb": "0", 
            "anonymous": "0",
            "stream": "0",
            "sd": "0",
            "keywords": "",
            "personal_release": "0",
            "internal": 0,
            "featured": 0,
            "free": 0,
            "doubleup": 0,
            "sticky": 0,
            "torrent-cover": '',
        }
`

Thank you !

**Comment by HDVinnie** [comment]
@mibj2 I dont know if there is language barrier or what but this Issue here is for a [Request] to add the ability to upload poster and backdrops via API for no_meta categories! AKA its not a thing yet. For the last time READ before posting. Your more then welcome to submit a PR to add this functionality or bounty it for someone else to do.

**Comment by mibj2** [comment]
Great! I can't wait to see it updated! You're a person of value! Wishing you good work !
HaVinnie the you’re the best!
bye

**Comment by Roardom** [comment]
Yep, I'd be happy to review a PR that implements this according to the conditions I laid out above.
